File: examples/python/06_multiple_outcomes.py
"""Simulated multiple-outcome CMR example."""

import numpy as np

import cmrdesign as cmr

rng = np.random.default_rng(106)
n_pilot = 800
d = np.r_[np.ones(n_pilot // 2), np.zeros(n_pilot // 2)]
y1 = np.r_[rng.beta(2, 5, n_pilot // 2), rng.beta(4, 4, n_pilot // 2)]
y2 = np.r_[rng.beta(3, 6, n_pilot // 2), rng.beta(5, 5, n_pilot // 2)]
y = np.c_[y1, y2]

fit = cmr.cmr_multiple_outcomes(
    y,
    d,
    weights=[0.4, 0.6],
    estimand="coprimary",
    alpha=0.05,
    method="bounded",
)

print({"pi_treatment": fit.pi, "certificate": fit.U_CMR})

File: examples/r/06_multiple_outcomes.R
# Simulated multiple-outcome CMR example.

library(cmrdesign)

set.seed(106)
n_pilot <- 800L
d <- c(rep(1, n_pilot / 2), rep(0, n_pilot / 2))
y1 <- c(rbeta(n_pilot / 2, 2, 5), rbeta(n_pilot / 2, 4, 4))
y2 <- c(rbeta(n_pilot / 2, 3, 6), rbeta(n_pilot / 2, 5, 5))
y <- cbind(primary = y1, secondary = y2)

fit <- cmr_multiple_outcomes(
  y,
  d,
  weights = c(primary = 0.4, secondary = 0.6),
  estimand = "coprimary",
  alpha = 0.05,
  method = "bounded"
)

print(list(pi_treatment = fit$pi, certificate = fit$U_CMR))
